src/components/UserSelector.tsx

```
@@ -1,16 +1,72 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import cn from 'classnames';
+
+type Props = {
+  users: User[];
+  onSelect: (userId: number) => Promise<void>;
+  selectedId: number | null;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  onSelect,
+  selectedId,
+}) => {
+  const [isActive, setIsActive] = useState(false);
+
+  const [selectedUserName, setSelectedUserName] = useState('');
+
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    function handleClickOutside(e: Event) {
+      if (!dropdownRef.current) {
+        return;
+      }
+
+      const target = e.target;
+
+      if (!(target instanceof Node)) {
+        return;
+      }
+
+      if (!dropdownRef.current.contains(target)) {
+        setIsActive(false);
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const toggleDropdown = () => setIsActive(prev => !prev);
+
+  const handleClick = (id: number, nameUser: string) => {
+    onSelect(id);
```

You call the async prop `onSelect(id)` without handling a possible rejection. Since `onSelect` returns a Promise, an error there could trigger an unhandled rejection. The checklist recommends using try/catch for async operations. Consider awaiting in a try/catch or adding `.catch` to make failures explicit and avoid silent errors (e.g., `try { await onSelect(id); } catch (e) { console.error(e); }`).